fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/JDBCIdentityCacheImpl.java

```diff
@@ -70,6 +72,30 @@ public Integer getResourceTypeId(String resourceType) throws FHIRPersistenceExce
         return result;
     }
 
+    @Override
+    public String getResourceTypeName(Integer resourceTypeId) throws FHIRPersistenceException {
+        String result = cache.getResourceTypeNameCache().getName(resourceTypeId);
+        if (result == null) {
+            // try the database instead and cache the result
+            Map<String, Integer> resourceMap = resourceDAO.readAllResourceTypeNames();
+            for (Map.Entry<String, Integer> entry : resourceMap.entrySet()) {
+                if (entry.getValue() == resourceTypeId) {
+                    result = entry.getKey();
+                    break;
+                }
+            }
+
+            if (result == null) {
+                // likely a configuration error, caused by the schema being generated
+                // for a subset of all possible resource types
+                throw new FHIRPersistenceDataAccessException("Resource type ID not registered in database: '" + resourceTypeId + "'");
+            }
+
+            cache.getResourceTypeNameCache().addEntry(resourceTypeId, result);
```

Comment:
Can we populate the cache with all the entries we get back from the DB, instead of just this one?  And even populate the getResourceTypeCache as well.  Same thing with getResourceTypeId method, can that populate this cache with at least the entry it gets back?

Reply:
done